cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ModelUtils.java

```diff
@@ -812,4 +813,62 @@ public static List<String> generateAllServiceDnsNames(String namespace, String s
 
         return dnsNames;
     }
+
+    /**
+     * Validate cpu and memory resources.
+     * Early resources validation avoids triggering any pod operation with invalid configuration.
+     * 
+     * For both cpu and memory, this method checks that request is greater than zero, 
+     * limit is greater than zero and limit is greater than or equal to the request.
+     * 
+     * @param resources Resources configuration.
+     * @param owner Resources owner.
+     */
+    public static void validateComputeResources(ResourceRequirements resources, String owner) {
+        List<String> errors = ModelUtils.validateComputeResources(resources, "cpu", owner);
+        errors.addAll(ModelUtils.validateComputeResources(resources, "memory", owner));
+        if (errors.size() > 0) {
+            throw new InvalidResourceException(errors.toString());
+        }
+    }
+
+    /**
+     * Validate compute resources for a given owner.
+     * 
+     * This method checks that request is greater than zero, limit is greater than zero 
+     * and limit is greater than or equal to the request.
+     * 
+     * @param resources Compute resources configuration.
+     * @param type Compute resources type.
+     * @param owner Compute resources owner.
+     * 
+     * @return Error set.
+     */
+    private static List<String> validateComputeResources(ResourceRequirements resources, String type, String owner) {
+        List<String> errors = new ArrayList<>();
+        String oid = owner != null && !owner.isBlank() ? owner : "component";
```

Comment:
What are the values when this would not be passed? It seems fairly unnecessary to me. It is a private method which I would expect to be always called in the right way.